src/main/java/graphql/schema/GraphQLArgument.java

```diff
@@ -116,6 +117,37 @@ public boolean hasSetValue() {
         return value;
     }
 
+    /**
+     * This static helper method will give out a java value based on the semantics captured
+     * in the {@link InputValueWithState} from {@link GraphQLArgument#getArgumentValue()}
+     *
+     * Note : You MUST only call this on a {@link GraphQLArgument} that is part of a fully formed schema.  We need
+     * all of the types to be resolved in order for this work correctly.
+     *
+     * @param argument the fully formed {@link GraphQLArgument}
+     * @param <T>      the type you want it cast as
+     *
+     * @return a value of type T which is the java value of the argument
+     */
+    public static <T> T getArgumentValue(GraphQLArgument argument) {
```

Comment:
related to the other comment: should we make this more clear? Like `getArgumentValueOrNull` or throw an Excpetion if the value is not set? Or adding an argument how to deal with not set values: like `boolean convertNotSetToNull`?